main.go

```diff
@@ -9,18 +9,41 @@ import (
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/aws/session"
 	"github.com/aws/aws-sdk-go/service/ecr"
+	"regexp"
+
+	"strings"
 )
 
+// TagFiltering type for image tag filtering
+type TagFiltering struct {
+	regExp string
```

Comment:
why not make this a compiled regex object?

Reply:
not necessary. I'm doing simple string match and for this, i don't need to compile it. 
In golang compiled regexp are used for more complicated actions: https://golang.org/pkg/regexp/#MatchString